opm/material/common/FastSmallVector.hpp

```diff
@@ -137,10 +139,46 @@ class FastSmallVector
     const ValueType& operator[](size_t idx) const
     { return dataPtr_[idx]; }
 
-    //! number of the element
-    size_t size() const
+    using size_type = typename std::vector<ValueType>::size_type;
+
+    //! number of elements
+    size_type size() const
     { return size_; }
 
+    //! Iterator type is a plain pointer, so be warned there is no validity checking.
+    using Iterator = ValueType*;
+
+    //! To support range-for etc.
+    Iterator begin() const
+    { return dataPtr_; }
+
+    //! To support range-for etc.
+    Iterator end() const
+    { return dataPtr_ + size_; }
```

Comment:
The `Iterator` type here might be a little misleading, depending on what `ValueType` is.  Suppose `ValueType` is something simple like `double`.  Then these functions return `double*`&ndash;i.e., pointer to mutable&ndash;from `const` member functions.  I think I'd prefer something like a `ConstIterator` for those cases, and that the regular `Iterator` be restricted to the non-`const` member functions.

Reply:
I have added ConstIterator, and made a complete set of [c]begin()/end() functions. Please verify that it is as expected.